docs/ProviderDesign.md

```diff
@@ -0,0 +1,79 @@
+# Provider's view of the FLOCX
+
+![ProviderDesign](../images/ProviderService-Marketplace.png)
+
+
+
+
+## Multi-tenant IRONIC:
+Ironic manages multiple pools of hardware each owned by a separate user/group/organization.
+
+
+## Provider Service:
+A centralized service per OpenStack deployment that manages offers and contracts 
+on behalf of all its tenants of the OpenStack.
+
+## Keystone:
+Provides authentication and authorization services.
+
+
+## Workflow:
+
+Following steps describe the process of offering physical hardware in the market by individual owners
+and receiving contracts from the marketplace.
+
+
+
+1. Hardware pool owners decide which nodes to offer in the marketplace and submit offers to the provider service
+2. Provider service confirms owner-hardware relationship before accepting offers. 
+3. For each accepted offer it fetches the hardware configuration from Ironic
+4. These offers are published to the marketplace as soon as possible. 
```

Comment:
It may be worth being a bit more precise here - it sounds like the provider will try and publish to the marketplace immediately, but if that fails then there'll be a background service that looks at all 'waiting' offers and tries to publish them periodically. Is this right?

Reply:
Elaborated to include the said idea.